src/pages/pu-vendors/form/VendorsForm.js

```diff
@@ -0,0 +1,293 @@
+import { Grid, FormControlLabel, Checkbox } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { RemittanceSettingsRepository } from 'src/repositories/RemittanceRepository'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { DataSets } from 'src/resources/DataSets'
+import { PurchaseRepository } from 'src/repositories/PurchaseRepository'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+import { SystemFunction } from 'src/resources/SystemFunction'
+
+export default function VendorsForm({ labels, maxAccess: access, recordId, setStore }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: RemittanceSettingsRepository.Profession.page
+  })
+
+  const { maxAccess, changeDT } = useDocumentType({
+    functionId: SystemFunction.Vendor,
+    access: access,
+    enabled: !recordId
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      reference: '',
+      name: '',
+      flName: '',
+      currencyId: '',
+      paymentMethod: '',
+      tradeDiscount: '',
+      keywords: '',
+      groupId: '',
+      status: '',
+      taxId: '',
+      taxRef: '',
+      isInactive: false,
+      isTaxable: false
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validate: values => {
+      const errors = {}
+
+      if (values.isTaxable && !values.taxRef) {
+        errors.taxRef = ' '
+      }
+
+      return errors
+    },
+    validationSchema: yup.object().shape({
+      name: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const response = await postRequest({
+        extension: PurchaseRepository.Vendor.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!obj.recordId) {
+        setStore({
+          recordId: response.recordId,
+          name: obj.name
+        })
+        toast.success(platformLabels.Added)
+
+        formik.setValues({
```

Comment:
please replace by  formik.setFieldValue('recordId',response.recordId) not need for destructuring